src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -619,18 +621,23 @@ pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
             if (!matched)
                 mismatch_slices.push_back(std::make_tuple(slice_node, slice_begin, slice_end));
         }
+        if (mismatch_slices.empty())
+            return true;
+
+        if (mismatch_slices.size() == slice_out_index_in_concat.size())
+            return false;
```

Comment:
I looks like we need to track if this `replace_output_update_name` call changed the graph or not
<img width="368" alt="{21E1FFBE-0E1E-427D-930B-E58AFE8419B3}" src="https://github.com/user-attachments/assets/baaf1c45-f3b8-41d8-936b-930a51396e11" />
and return true/false depending on that here
<img width="297" alt="{542A918E-AA47-4161-B2D4-E9CB677DDE4B}" src="https://github.com/user-attachments/assets/f919098a-36ab-4ff6-ae0e-791f65b324e6" />